docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -39,12 +39,30 @@ impl Storage {
 If you have defined a `Storage` struct following this naming scheme, then it will be made available to you through the reserved `storage` keyword within your contract functions.
 
 :::warning Using slot `0` is not supported!
-No storage values should be initialized at slot `0`. This is a known issue that will be fixed in the future.
+No storage values should be initialized at slot `0` - storage slots begin at `1`. This is a known issue that will be fixed in the future.
 :::
 
+## Storage Slots
+
+Internally, Aztec keeps contract state in a storage slot key value store of type `Map<Field, Field>`, stored as a merkle tree(?). For now, storage slot positions for each variable must be explicitly assigned inside the `Storage impl`. Except for `Map` types, storage is in contiguous blocks, so you must calculate how many slots each public variable requires, and increment the next variable's slot by this amount.
+
+In defining contract storage, `Map`s can be treated as occupying only 1 storage slot (its "base_slot"), because the actual values are stored in derived slots calculated as `pedersen_hash(base_slot, key)`.
+
+:::info
+Private variables only require one single slot, because all notes are linked to contract state through a single `storage slot` attribute in their header.
+:::
+
+We currently do not support any "packing" type optimizations as in the EVM.
```

Comment:
nit: This isnt a property of the EVM, rather an optimisation implemented by its languages.

Reply:
whoops, danke!